spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -68,9 +69,13 @@ public SimpleRouteLocator(String servletPath, ZuulProperties properties) {
 		if (this.routes.get() == null) {
 			this.routes.set(locateRoutes());
 		}
+		return assembleRoutes(Collections.unmodifiableMap(this.routes.get()));
```

Comment:
Maybe it also makes sense to add test that if somebody tries to modify routes inside assembleRoutes it will fail with error? wdyt?